src/base/QXmppTrustMessageKeyOwner.cpp

```diff
@@ -0,0 +1,176 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppTrustMessageKeyOwner.h"
+
+#include <QDomElement>
+
+#include "QXmppConstants_p.h"
+#include "QXmppUtils.h"
+
+///
+/// \class QXmppTrustMessageKeyOwner
+///
+/// \brief The QXmppTrustMessageKeyOwner class represents a key owner of the
+/// trust message as defined by \xep{0434, Trust Messages (TM)}.
+///
+/// \since QXmpp 1.5
+///
+
+class QXmppTrustMessageKeyOwnerPrivate : public QSharedData
+{
+public:
+    QString jid;
+    QList<QString> trustedKeys;
+    QList<QString> distrustedKeys;
+};
+
+///
+/// Constructs a trust message key owner.
+///
+QXmppTrustMessageKeyOwner::QXmppTrustMessageKeyOwner()
+    : d(new QXmppTrustMessageKeyOwnerPrivate)
+{
+}
+
+///
+/// Constructs a copy of \a other.
+///
+/// \param other
+///
+QXmppTrustMessageKeyOwner::QXmppTrustMessageKeyOwner(const QXmppTrustMessageKeyOwner &other) = default;
+
+QXmppTrustMessageKeyOwner::~QXmppTrustMessageKeyOwner() = default;
+
+///
+/// Assigns \a other to this trust message key owner.
+///
+/// \param other
+///
+QXmppTrustMessageKeyOwner &QXmppTrustMessageKeyOwner::operator=(const QXmppTrustMessageKeyOwner &other) = default;
+
+///
+/// Returns the bare JID of the key owner.
+///
+/// \return the key owner's bare JID
+///
+QString QXmppTrustMessageKeyOwner::jid() const
+{
+    return d->jid;
+}
+
+///
+/// Sets the bare JID of the key owner.
+///
+/// If a full JID is passed, it is converted into a bare JID.
+///
+/// \param jid key owner's bare JID
+///
+void QXmppTrustMessageKeyOwner::setJid(const QString &jid)
+{
+    d->jid = QXmppUtils::jidToBareJid(jid);
+}
+
+///
+/// Returns the IDs of the keys that are trusted.
+///
+/// \return the IDs of trusted keys
+///
+QList<QString> QXmppTrustMessageKeyOwner::trustedKeys() const
+{
+    return d->trustedKeys;
+}
+
+///
+/// Sets the IDs of keys that are trusted.
+///
+/// \param keyIds IDs of trusted keys
+///
+void QXmppTrustMessageKeyOwner::setTrustedKeys(const QList<QString> &keyIds)
+{
+    d->trustedKeys = keyIds;
+}
+
+///
+/// Returns the IDs of the keys that are distrusted.
+///
+/// \return the IDs of distrusted keys
+///
+QList<QString> QXmppTrustMessageKeyOwner::distrustedKeys() const
+{
+    return d->distrustedKeys;
+}
+
+///
+/// Sets the IDs of keys that are distrusted.
+///
+/// \param keyIds IDs of distrusted keys
+///
+void QXmppTrustMessageKeyOwner::setDistrustedKeys(const QList<QString> &keyIds)
+{
+    d->distrustedKeys = keyIds;
+}
+
+void QXmppTrustMessageKeyOwner::parse(const QDomElement &element)
```

Comment:
Please fix:
```
/home/lnj/dev/qxmpp/src/base/QXmppTrustMessageElement.cpp:145: warning: documented symbol 'void QXmppTrustMessageElement::parse' was not declared or defined.
/home/lnj/dev/qxmpp/src/base/QXmppTrustMessageElement.cpp:161: warning: no matching class member found for 
  void QXmppTrustMessageElement::toXml(QXmlStreamWriter *writer) const

/home/lnj/dev/qxmpp/src/base/QXmppTrustMessageKeyOwner.cpp:134: warning: documented symbol 'void QXmppTrustMessageKeyOwner::parse' was not declared or defined.
/home/lnj/dev/qxmpp/src/base/QXmppTrustMessageKeyOwner.cpp:149: warning: no matching class member found for 
  void QXmppTrustMessageKeyOwner::toXml(QXmlStreamWriter *writer) const
```

... by adding `/// \cond` & `/// \endcond` like in (all) other files.